### PR TITLE
Update discord.py to current version

### LINKED ISSUE #24: Update discordpy to a supported version
datcord_bot has been stuck on an old discordpy version for years. It would be nice to upgrade it to the latest version, so that the bot can run with Python 3.7 and later.

### PR DESCRIPTION
**Pull request by winston-yallow**
This updates discord.py to version 1.2 which works with Python 3.7.

closes #24

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -21,6 +21,6 @@ keywords = [
 [tool.poetry.dependencies]
 # Python 3.7 and later aren't working due to discord.py < 1.0.0 being used
 # See <https://github.com/Rapptz/discord.py/issues/1396>
-python = "^3.5, < 3.7"
-"discord.py" = "^0.16"
+python = "^3.5.3, < 3.8"
```
Comment (Calinou) [inline pyproject.toml]: If Python 3.8 can run the bot, feel free to remove the upper version range check :slightly_smiling_face: 

Also remove the comment above, as it's no longer relevant.

Comment (winston-yallow) [reply pyproject.toml]: Comment is removed.
Regarding py 3.8: I will check if it works after I have the bot running again with discord.py 1.2

**Comment by winston-yallow** [comment]
I could use some pointers why the ci is failing. I reformatted the code locally with black but the ci is still complaining. Are there any more detailed logs than just the info that black failed?

**Comment by Calinou** [comment]
@winston-yallow Running `black main.py --diff`, I get:

```diff
❯ black main.py --diff
--- main.py     2019-10-30 12:44:13.442742 +0000
+++ main.py     2019-10-30 12:44:31.252554 +0000
@@ -504,11 +504,11 @@
                         if role not in message.author.roles:
                             await message.author.add_roles(role)
                             tmp = await message.channel.send(
                                 ":white_check_mark: User {0} added to {1}.".format(
                                     message.author.name, role.name
-                                ),
+                                )
                             )
                             await delete_edit_timer(
                                 tmp, FEEDBACK_DEL_TIMER, call_msg=message
                             )
                         else:
@@ -518,19 +518,19 @@
                             await delete_edit_timer(
                                 tmp, FEEDBACK_DEL_TIMER, error=True, call_msg=message
                             )
                     else:
                         tmp = await message.channel.send(
-                            ":no_entry: *You're not allowed to assign yourself to that role.*",
+                            ":no_entry: *You're not allowed to assign yourself to that role.*"
                         )
                         await delete_edit_timer(
                             tmp, FEEDBACK_DEL_TIMER, error=True, call_msg=message
                         )
                     break
             else:
                 tmp = await message.channel.send(
-                    ":no_entry: **{0}** <- *Role not found.*".format(new_role.upper()),
+                    ":no_entry: **{0}** <- *Role not found.*".format(new_role.upper())
                 )
                 await delete_edit_timer(
                     tmp, FEEDBACK_DEL_TIMER, error=True, call_msg=message
                 )
 
@@ -567,11 +567,11 @@
                     break
             else:
                 tmp = await message.channel.send(
                     ":no_entry: **{0}** <- You don't have that role.".format(
                         old_role.upper()
-                    ),
+                    )
                 )
                 await delete_edit_timer(
                     tmp, FEEDBACK_DEL_TIMER, error=True, call_msg=message
                 )
 ```

It seems this pre-commit hook isn't designed for CI use as it doesn't report diffs when failing. I solved this in `master` by calling it manually beforehand: https://github.com/Calinou/datcord_bot/commit/ff280a531a7f0fb89e3f1e7f84b31e22097e9ae0

As for the different formatting when using Black locally, I don't know about that, sorry. Make sure you're using the latest version of Black (19.10b0 as of writing).

**Comment by winston-yallow** [comment]
Thanks for the diff. I will need to check why it didn't work locally as expected. The version is the correct one. Maybe I was just too tired yesterday...

Black seems to pass now. However the `black *.py --diff` is throwing an error (127).

I still need to change a few things (for example how sending files works) before this is ready to merge.

**Comment by winston-yallow** [comment]
Everything seems to work now, will test py 3.8 tomorrow.

File: main.py [hunk 2]
```diff
@@ -453,8 +456,8 @@ def populate_memes() -> None:
                 s += ", "
         s += "```"
 
-        await client.send_message(message.channel, s)
-        await client.delete_message(message)
```
Comment (winston-yallow) [inline main.py:457]: I simply kept this, but I am wondering why the original message is deleted? Also this only seems to work with the "Manage Messages" permission additionally to the ones you mentioned.

**Comment by winston-yallow** [comment]
Except the one comment about the msg deletion this is good by me. I double checked the migration guide from discord.py and tested this with python 3.8 today. All commands work fine.

File: main.py (hunk 2 above)
Comment (Calinou) [reply main.py:457]: I think the idea was to remove confirmation messages printed by the bot after a few seconds. I don't think it's worth the added complexity, so feel free to remove that feature.

**Comment by Calinou** [comment]
I tested this PR locally and it seems to work well, so I'll merge it. Thanks a lot for your contribution :slightly_smiling_face:

**Pull request merged**